### PR TITLE
Implemented remind function

### PR DESCRIPTION
**Pull request by Roahhh**
Ready for review

### CODE REVIEW

File: src/main/java/linenux/command/RemindCommand.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+	@Override
+	public CommandResult execute(String userInput) {
+		assert userInput.matches(REMIND_PATTERN);
+		assert this.schedule != null;
+
+		String argument = extractArgument(userInput);
+
+		Either<String, CommandResult> keywords = extractKeywords(argument);
+
+        if (keywords.isRight()) {
```
Comment (cadmusthefounder) [inline src/main/java/linenux/command/RemindCommand.java]: indentation is off

File: src/main/java/linenux/command/RemindCommand.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        	return makeResult(task, actualReminder);
+        } else {
+        	this.requiresUserResponse = true;
+        	this.foundTasks = tasks;
+        	this.reminderForPrompt = actualReminder;
+        	return makePromptResult(tasks);
+        }
+	}
+
+	private String extractArgument(String userInput) {
```
Comment (cadmusthefounder) [inline src/main/java/linenux/command/RemindCommand.java]: I think your eclipse nvr set the tab as space

Comment (cadmusthefounder) [inline src/main/java/linenux/command/RemindCommand.java]: Oh and also rmb to re-set the remove white space when saving the file

File: src/test/java/linenux/command/RemindCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+
+/**
+ * JUnit test for remind command.
+ */
+public class RemindCommandTest {
+	private Schedule schedule;
+	private RemindCommand remindCommand;
+
+	@Before
+	public void setupRemindCommand() {
```
Comment (cadmusthefounder) [inline src/test/java/linenux/command/RemindCommandTest.java]: I am not sure there is a need to test for deadline, todo and event separately unless there is some functionality difference (Like cannot add reminder after deadline and event end time). Else I think just testing for a generic task should be fine

File: src/test/java/linenux/command/RemindCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+		ArrayList<Task> tasks = new ArrayList<>();
+		tasks.add(new Task("Todo1"));
+		tasks.add(new Task("Todo2"));
+		tasks.add(new Task("Deadline1", LocalDateTime.of(2016, 1, 1, 1, 0)));
+		tasks.add(new Task("Deadline2", LocalDateTime.of(2016, 1, 1, 13, 0)));
+		tasks.add(new Task("Event1", LocalDateTime.of(2016, 1, 1, 1, 0), LocalDateTime.of(2016, 1, 1, 13, 0)));
+		tasks.add(new Task("Event2", LocalDateTime.of(2016, 1, 1, 13, 0), LocalDateTime.of(2016, 1, 1, 23, 0)));
```
Comment (yihangho) [inline src/test/java/linenux/command/RemindCommandTest.java]: Can you keep track of each instance of `Task` in an instance variable? The current setup is too fragile. For e.g., if I add

```java
tasks.add(new Task("Todo0"));
```

before line 31, more than half of the test cases here will fail.

File: src/main/java/linenux/command/RemindCommand.java [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+	}
+
+	@Override
+	public CommandResult execute(String userInput) {
+		assert userInput.matches(REMIND_PATTERN);
+		assert this.schedule != null;
+
+		String argument = extractArgument(userInput);
+
+		Either<String, CommandResult> keywords = extractKeywords(argument);
```
Comment (Roahhh) [inline src/main/java/linenux/command/RemindCommand.java]: Need to refactor the code. As per discussion

**Pull request merged**